modules/calib3d/src/fundam.cpp

```diff
@@ -1224,7 +1224,10 @@ double cv::sampsonDistance(InputArray _pt1, InputArray _pt2, InputArray _F)
 {
     CV_INSTRUMENT_REGION();
 
-    CV_Assert(_pt1.type() == CV_64F && _pt2.type() == CV_64F && _F.type() == CV_64F);
+    CV_Assert(((_pt1.type() & CV_MAT_DEPTH_MASK) == CV_64F) &&
+              ((_pt2.type() & CV_MAT_DEPTH_MASK) == CV_64F) &&
+              ((_F.type() & CV_MAT_DEPTH_MASK) == CV_64F));
```

Comment:
Please use Mat::depth() instead of bit magic: https://docs.opencv.org/4.x/d3/d63/classcv_1_1Mat.html#a8da9f853b6f3a29d738572fd1ffc44c0